TaskModules/powershell/TaskModuleIISManageUtility/AppCmdOnTargetMachines.ps1

```diff
@@ -731,6 +731,35 @@ function Start-Stop-Recycle-ApplicationPool {
     Invoke-VstsTool -Filename $appCmdPath -Arguments $appCmdArgs -RequireExitCodeZero
 }
 
+function ConfigureWebsiteAuthentication {
+    param (
+        [string]$basicAuthentication,
+        [string]$windowsAuthentication,
+        [string]$websiteName
+    )
+
+    Write-Verbose "Configuring website authentication"
+    $appCmdPath, $iisVersion = Get-AppCmdLocation -regKeyPath $AppCmdRegKey
+
+    $anonymousAuthentication = $basicAuthentication -ieq "false" -and $windowsAuthentication -ieq "false"
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:anonymousAuthentication /enabled:{1} /commit:apphost', $websiteName, $anonymousAuthentication)

```

Comment:
for enabling basic and windows we need to turn off anonymous auth. https://www.youtube.com/watch?v=fZ2GoVnUh6k